src/app/merge-duplicates/merge-duplicates-table/merge-duplicates-table.component.html

```diff
@@ -0,0 +1,89 @@
+<div class="merge-duplicates-table-wrapper" *ngIf="mergeDuplicates">
+
+  <table mat-table [dataSource]="processTableData()">
```

Comment:
this will trigger with every change detection cycle, rebuilding the array over and over again;
i think it would be better to turn @Input mergeDuplicates into a setter and make the processing there; then you can use mergeDuplicates as a data source